ros2_generate_interface_docs/ros2_generate_interface_docs/utils.py

```diff
@@ -0,0 +1,305 @@
+
+
+def write_template(template_name, data, output_file):
+    """
+    Write the data in the template.
+
+    Parameters
+    ----------
+    template_name: str
+        name of the template to write
+    data:
+        data that is used to fill the template
+    output_file: str
+        path where the file will be written
+
+    """
+    msg_index_template, template_path = load_template(template_name)
+    output = StringIO()
+    interpreter = em.Interpreter(
+        output=output,
+        options={
+            em.BUFFERED_OPT: True,
+            em.RAW_OPT: True,
+        },
+    )
+    try:
+        with open(template_path, 'r') as h:
+            template_content = h.read()
+            interpreter.invoke(
+                'beforeFile', name=template_name, file=h, locals=data)
+        interpreter.string(template_content, template_path, locals=data)
+        interpreter.invoke('afterFile')
+    except Exception as e:  # noqa: F841
+        print(f"{e.__class__.__name__} when expanding '{template_name}' into "
+              f"'{output_file}': {e}", file=sys.stderr)
+        raise
+
+    content = output.getvalue()
+    interpreter.shutdown()
+
+    with open(os.path.join(output_file), 'w') as f:
+        f.write(content)
+
+
+def generate_compact_definition(imported_interface, indent, get_slot_types):
+    """
+    Create the compact definition dictionary.
+
+    This function remove all the common dict keys to get all the contants in the message
+    dictionary. Then it will generate a dictionary will the contants and field of the message.
+    If the message is based in other message, then this text will contain a link to this
+    interface.
+
+    Parameters
+    ----------
+    imported_interface: obj
+        class with all the data about the interface
+    indent: int
+        number of indentations to add to the generated text
+    get_slot_types: function
+        function that returns an OrderedDict of the slot types of a message.
+
+    Returns
+    -------
+    compact: dict
+        Dictionary with the compact definition (constanst and message with links)
+
+    """
+    ignored_keys_ = IGNORED_KEYS
+    ignored_keys_ += list(imported_interface.get_fields_and_field_types().keys())
+    ignored_keys_ += ['_'+x for x in imported_interface.get_fields_and_field_types().keys()]
+    compact = {'constant_types': [],
+               'constant_names': [],
+               'links': [],
+               'field_types': [],
+               'field_names': []}
+    for key in imported_interface.__dict__.keys():
+        if key not in ignored_keys_:
+            compact['constant_types'].append(key)
+            compact['constant_names'].append(getattr(imported_interface, key))
+
+    fields = imported_interface.get_fields_and_field_types()
+    for field, field_types in fields.items():
+        if(field_types not in BASIC_TYPES and
+           'string' not in field_types):
```

Comment:
This line is not visually distinguishable from the next one. Wrap the code like this instead:

```
if (
    ... and
    ...
):
    ...
```